test/test_response.py

```diff
@@ -198,7 +200,9 @@ def test_io(self):
         self.assertRaises(IOError, resp3.fileno)
 
     def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+        sock = socket.socket()
+        self.addCleanup(sock.close)
```

Comment:
That just shouldn't happen: there should be no other references to the socket keeping it alive. =( Any idea where the other reference is?